cmake/CompilerSetup.cmake

```diff
@@ -67,6 +71,13 @@ if(CLANG OR GNU)
     )
   endif()
 
+  if(NINJA)
+    # If building under Ninja, disable tty detection and force color output
+    if(CLANG OR GNU)
+      list(APPEND common_flags -fdiagnostics-color)
```

Comment:
fwiw, this doesn't work for me. I can see (via `ninja -v`) that the flag *does* get passed, but no colour is present.

But I might actually prefer it that way.

I do like colourized output, but I also like being able to pipe it to other programs; colours sometimes mess that up. tty detection is often a good compromise that easily allows both.

But I'm not picky here. If it works for you, then I have no objections to keeping it. (I should also point out I typically/always use the make generator, so have minimal/no experience with ninja. Maybe I'm just using it wrong.)

No action required.